src/code/code_800EC960.c

```diff
@@ -1207,917 +1257,1096 @@ u8 D_8016B9F3;
+                                   &D_801333E8);
+        } else if ((sPrevOcarinaNoteIdx != NOTE_NONE) && (sCurOcarinaNoteIdx == NOTE_NONE)) {
+            // cancels ocarina for a non-valid input
             Audio_StopSfxById(NA_SE_OC_OCARINA);
         }
     }
 }
 
-void func_800ED848(u8 inputEnabled) {
-    sOcarinaInpEnabled = inputEnabled;
+/**
+ * Directly enable the ocarina to receive input without
+ * properly resetting it based on an ocarina instrument id
+ * Unused.
+ */
+void AudioOcarina_EnableInput(u8 inputEnabled) {
+    sIsOcarinaInputEnabled = inputEnabled;
 }
 
-void Audio_OcaSetInstrument(u8 arg0) {
-    if (D_80130F10 == arg0) {
+/**
+ * Resets ocarina properties based on the ocarina instrument id
+ * If ocarina instrument id is "OCARINA_INSTRUMENT_OFF", turn off the ocarina
+ * For all ocarina instrument ids, turn the ocarina on with the instrument id
+ */
+void AudioOcarina_Reset(u8 ocarinaInstrumentId) {
+    if (sOcarinaInstrumentId == ocarinaInstrumentId) {
         return;
     }
 
-    Audio_SeqCmd8(SEQ_PLAYER_SFX, 1, SFX_PLAYER_CHANNEL_OCARINA, arg0);
-    D_80130F10 = arg0;
-    if (arg0 == 0) {
-        sCurOcarinaBtnPress = 0;
-        sPrevOcarinaBtnPress = 0;
-        D_8016BA18 = 0;
-        D_8016BA10 = 0xFFFF;
-        func_800ED458(0);
+    Audio_SeqCmd8(SEQ_PLAYER_SFX, 1, SFX_CHANNEL_OCARINA, ocarinaInstrumentId);
+    sOcarinaInstrumentId = ocarinaInstrumentId;
+    if (ocarinaInstrumentId == OCARINA_INSTRUMENT_OFF) {
+        sOcarinaInputButtonCur = 0;
+        sOcarinaInputButtonPrev = 0;
+        sOcarinaInputButtonPress = 0;
+
+        sOcarinaInputButtonStart = 0xFFFF;
+
+        AudioOcarina_PlayControllerInput(false);
         Audio_StopSfxById(NA_SE_OC_OCARINA);
         Audio_SetSoundBanksMute(0);
-        sPlaybackState = 0;
-        sStaffPlaybackPos = 0;
-        sOcarinaInpEnabled = 0;
-        D_80130F3C = 0;
-        Audio_ClearBGMMute(SFX_PLAYER_CHANNEL_OCARINA);
+        sDisplayedState = 0;
+        sDisplayedStaffPos = 0;
+        sIsOcarinaInputEnabled = false;
+        sOcarinaFlags = 0;
+        // return to full volume for players 0 and 3 (background bgm) after ocarina is finished
+        Audio_ClearBGMMute(SFX_CHANNEL_OCARINA);
     } else {
-        sCurOcarinaBtnPress = 0;
-        Audio_GetOcaInput();
-        D_8016BA10 = sCurOcarinaBtnPress;
-        Audio_QueueSeqCmdMute(SFX_PLAYER_CHANNEL_OCARINA);
+        sOcarinaInputButtonCur = 0;
+        AudioOcarina_ReadControllerInput();
+        // Store button used to turn on ocarina
+        sOcarinaInputButtonStart = sOcarinaInputButtonCur;
+        // lowers volumes of players 0 and 3 (background bgm) while playing ocarina
+        Audio_QueueSeqCmdMute(SFX_CHANNEL_OCARINA);
     }
 }
 
-void Audio_OcaSetSongPlayback(s8 songIdxPlusOne, s8 playbackState) {
+void AudioOcarina_SetDisplayedSong(s8 songIdxPlusOne, s8 displayedState) {
```

Comment:
why is this called "displayed"? I can see code for playing the song, but "display" makes me think there should be code for drawing too

Reply:
think display in this context is just referring to what its showing the player. dont think that term has to mean drawing in all contexts.
That said, just going off of the contexts its used in (disclaimer: not reading much of the code yet), Id say the existing name calling it "playback" is pretty good. Its playing back the song for you as opposed to playing it. How come you wanted to move away from that name?

Reply:
The idea was that there were three staff  names introduced in message: the "playing staff", the "displayed staff" and "the recording staff". And the term "displayed" and "playback" were overlapping and often represented the same thing. So I ended up consolidating them into "displayed". But I wanted to read more of z_message after Dragorns comment to understand the system better and perhaps use "playback" as the go-to term for this.

Reply:
Reading through it now, I think I am starting to prefer the term "playback" over "displayed" as suggested as well

Reply:
cool.
yeah i think in general "display" doesnt necessarily have to imply doing the literal drawing. That would more traditionally have Draw in the name.
But yeah if playback makes sense in this context i think that will be more clear either way so its a win/win